### PR TITLE
refactor(rust): change uses of `ockam_vault_core::Foo` to use `ockam_core::vault::Foo` across crates

### LINKED ISSUE #2288: Move the `ockam_vault_core` create into `ockam_core`.
We're trying to reduce the number of crates, and part of that is moving `ockam_vault_core` into a new `vault` submodule of `ockam_core`. I think this basically looks like

1. Make a new `pub mod vault;` in [`ockam_core`](https://github.com/ockam-network/ockam/blob/develop/implementations/rust/ockam/ockam_core/src/lib.rs).
    - Note: unlike the [other modules](https://github.com/ockam-network/ockam/blob/develop/implementations/rust/ockam/ockam_core/src/lib.rs#L54-L60), I think we shouldn't `pub use vault::*` from the root here, since we'd like to move away from this.

2. Move the code from `ockam_vault_core/src/*` to a new `ockam_core/src/vault/*` directory (and updating anything necessary, renaming lib.rs to mod.rs, renaming any references that were previously `crate::Something` to be `crate::vault::Something`)

3. Update the [doc comments](https://github.com/ockam-network/ockam/blob/develop/implementations/rust/ockam/ockam_vault_core/src/lib.rs#L3) so that it says "this module" rather than "this crate" and such. (Ask if you need help wording this).

4. Change uses of `ockam_vault_core::Foo` to use `ockam_core::vault::Foo` across our crates.

5. Delete the `ockam_vault_core` crate, and any references to it (be sure to check `.github/workflows`, and to do a search for `ockam_vault_core`).

That's the basic idea anyway (I may have missed something). Feel free to ask questions if anything is unclear.

### PR DESCRIPTION
**Pull request by totsteps**
closes #2288

### Tasks

- [x] Make a new `pub mod vault;` in `ockam_core`.
- [x] Move code from `ockam_vault_core/src/*` to a new `ockam_core/src/vault/*` directory.
- [x] Update the doc comments.
- [x] Change uses of `ockam_vault_core::Foo` to `use ockam_core::vault::Foo` across crates.
- [x] Delete `ockam_vault_core` crate, and any references to it (be sure to check `.github/workflows`, and to do a search for `ockam_vault_core`).

### Checks
- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits).
- [x] All commits in this Pull Request follow the Ockam [commit message convention](https://www.ockam.io/learn/how-to-guides/contributing/CONTRIBUTING#commit-messages).
- [x] I accept the Ockam Community [Code of Conduct](https://www.ockam.io/learn/how-to-guides/high-performance-team/conduct/).
- [x] I have accepted the Ockam [Contributor Licence Agreement](https://www.ockam.io/learn/how-to-guides/contributing/cla/) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/ockam-network/contributors/blob/master/CONTRIBUTORS.csv) file in a separate pull request to the [ockam-network/contributors](https://github.com/ockam-network/contributors) repository.

### CODE REVIEW

File: implementations/rust/ockam/ockam_core/src/vault/types.rs [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+/// P256 private key length
+pub const P256_SECRET_LENGTH: usize = 32;
+/// P256 public key length
+pub const P256_PUBLIC_LENGTH: usize = 65;
+/// AES256 private key length
+pub const AES256_SECRET_LENGTH: usize = 32;
+/// AES128 private key length
+pub const AES128_SECRET_LENGTH: usize = 16;
+
+cfg_if! {
```
Comment (SanjoDeundiak) [inline implementations/rust/ockam/ockam_core/src/vault/types.rs:18]: @antoinevg I think it's outdated approach, should we instead use compat?
P.S. This is out of scope of this PR

File: implementations/rust/ockam/ockam_core/src/vault/mod.rs [hunk 2]
```diff
@@ -0,0 +1,37 @@
+//! Core types and traits of the Ockam vault.
+//!
+//! This crate contains the core types and traits of the Ockam vault and is intended
+//! for use by other crates that either provide implementations for those traits,
+//! or use traits and types as an abstract dependency.
+#![deny(unsafe_code)]
```
Comment (SanjoDeundiak) [inline implementations/rust/ockam/ockam_core/src/vault/mod.rs:6]: I think 12 lines below should be deleted, as all of this should be already present in ockam_core/src/lib.rs

**Review by SanjoDeundiak — CHANGES_REQUESTED** [review]
Looks great. One small issue to fix

**Comment by mrinalwadhwa** [comment]
@totsteps awesome, this looks close to ready for merging after a couple of quick changes.


1• rebase to latest `develop` - when you do this, most file deletions `.github/workflows` will become unnecessary since ffdc3eef56b7369efe5b45b745843e88b8f730a7 removed them a few days ago.

---
2• our commitlinter will complain about this commit message because it enforces lower case.
```
chore(rust): change uses of ockam_vault_core::Foo to use ockam_core::vault::Foo across crates
```
Changing the message to this will fix the problem:
```
chore(rust): change uses of `ockam_vault_core::Foo` to use `ockam_core::vault::Foo` across crates
```
A better commit type for all commits in this PR is `refactor:` so I suggest changing the message to:
```
refactor(rust): change uses of `ockam_vault_core::Foo` to use `ockam_core::vault::Foo` across crates
```

**Comment by totsteps** [comment]
Somehow I messed up when rebasing!

Will open another PR.

**Pull request closed without merging**